tripy/docs/post0_developer_guides/integrating-tensorrt-python-plugins.md

```diff
@@ -0,0 +1,125 @@
+) -> Tuple[
+    Union[str, bytes], Union[str, bytes], trtp.KernelLaunchParams, trtp.SymExprs
+]:
+    """
+    Implements the plugin using a Triton kernel.
+    """
+    src = triton.compiler.ASTSource(
+        fn=add_kernel,
+        signature="fp32,i32,fp32",
+        constants={
+            "BLOCK_SIZE": block_size,
+        },
+    )
+
+    compiled_kernel = triton.compile(src)
+    launch_params = trtp.KernelLaunchParams()
+
+    # Set the grid, block dims and shared memory for the kernel (as symbolic expressions)
+    N = inp0.shape_expr.numel()
+    launch_params.grid_x = trtp.cdiv(N, block_size)
+    launch_params.block_x = compiled_kernel.metadata.num_warps * 32
+    launch_params.shared_mem = compiled_kernel.metadata.shared
+
+    # Define extra scalar arguments for the kernel (as symbolic expressions)
+    extra_args = trtp.SymIntExprs(1)
+    extra_args[0] = trtp.SymInt32(N)
+
+    return (
+        compiled_kernel.metadata.name,
+        compiled_kernel.asm["ptx"],
+        launch_params,
+        extra_args,
+    )
+```
+
+### Using the Plugin in Tripy
+
+Once we have defined our plugin, it is accessible under `trt.plugin.op.<namespace>.<plugin_name>`, which is `trt.plugin.op.example.elemwise_add_plugin` for the above plugin. `tp.plugin` can be used to add the plugin to Tripy.
+
+```py
+inp = tp.iota((2, 2))
+out = tp.plugin(
+    trt.plugin.op.example.elemwise_add_plugin,
+    [inp],  
+    block_size = 256,
+)
+
+assert cp.allclose(cp.from_dlpack(out), cp.from_dlpack(inp + 1))
+```
+
+### Supported attribute types
+
+QDPs support either scalar attributes of selected primitive Python types or 1-D Numpy arrays of selected Numpy data types. For Numpy array attributes, the corresponding argument in `@trtp.register` must be annotated with `numpy.typing.NDArray[dtype]`, where `dtype` is the expected Numpy data type. Please refer to the [TensorRT Plugin Guide](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/pluginGuide.html) for details on exact primitive and Numpy data types supported by QDPs.
+
+## Understanding the Implementation
+
+### Plugin Registration
+
+The `@trtp.register` decorator registers our plugin with TensorRT. The plugin name follows the format 
+`namespace::plugin_name`. The registration function:
+
+1. Takes input tensor descriptors and plugin attributes as arguments
+2. Returns output tensor descriptors
+3. Defines the plugin interface that Tripy will use
+
+### Plugin Implementation 
+
+The `@trtp.aot_impl` decorator returns an ahead-of-time (AOT) compiled kernel, along with information required to invoke the kernel such as kernel launch parameters and extra scalar arguments. This plugin uses OpenAI Triton to define the kernel and uses its AOT compilation capabilities to retrieve PTX for the kernel.
+
+### Tripy Integration
+
+Tripy's `tp.plugin()` function accepts:
+- The plugin op
+- A list of input tensors
+- Plugin attributes as keyword arguments
+
+Links:
```

Comment:
Can put these under a seealso block:
```
:::{seealso}
...
:::
```

Reply:
Is `seealso` supported in Markdown?

Reply:
Not sure. We use the Myst parser to parse these guides before they are finally rendered, so our documentation supports a bunch of extra markup compared to Markdow.

Reply:
Looks like it is supported: https://myst-parser.readthedocs.io/en/v0.17.0/_sources/api/parsers.md.txt